### DISCUSSION TITLE
Failed to enable TLSv1.0 / TLSv1.1

### DISCUSSION THREAD

**Discussion by immortalua** [nginx-proxy/nginx-proxy#2483] [Q&A] [answered]
Hello. I need to use all versions of TLS in my deployment because my service uses many xp/win7 desktops etc.

I use default nginxproxy/nginx-proxy and nginxproxy/acme-companion Docker images

1. I enabled  SSL_POLICY=Mozilla-Old option for my backend 
And I see right settings in /etc/nginx/conf.d/default.conf
```nginx
ssl_protocols TLSv1 TLSv1.1 TLSv1.2 TLSv1.3;
ssl_ciphers 'ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384:DHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA:ECDHE-RSA-AES256-SHA:DHE-RSA-AES128-SHA256:DHE-RSA-AES256-SHA256:AES128-GCM-SHA256:AES256-GCM-SHA384:AES128-SHA256:AES256-SHA256:AES128-SHA:AES256-SHA:DES-CBC3-SHA';
ssl_prefer_server_ciphers on;
```

2. I checked for  openssl3 active ciphers "**openssl ciphers -v**" and there I found TLS1 ciphers

3. I tried to add in /etc/ssl/openssl.cnf these settings
```ini
[ssl_configuration]
system_default = tls_system_default
[tls_system_default]
MinProtocol = TLSv1
```

But all these actions did not lead to the desired result- enable TLSv1.0 TLSv1.1 TLSv1.2 TLSv1.3 on my deployment.

openssl s_client -connect backend_ext_url:443 -tls1
openssl s_client -connect backend_ext_url:443 -tls1_1

```
CONNECTED(00000005)
140599712252352:error:1409442E:SSL routines:ssl3_read_bytes:tlsv1 alert protocol version:../ssl/record/rec_layer_s3.c:1528:SSL alert number 70
---
no peer certificate available
---
No client certificate CA names sent
---
SSL handshake has read 7 bytes and written 135 bytes
Verification: OK
---
New, (NONE), Cipher is (NONE)
Secure Renegotiation IS NOT supported
Compression: NONE
Expansion: NONE
No ALPN negotiated
SSL-Session:
    Protocol  : TLSv1
    Cipher    : 0000
    Session-ID: 
    Session-ID-ctx: 
    Master-Key: 
    PSK identity: None
    PSK identity hint: None
    SRP username: None
    Start Time: 1718307803
    Timeout   : 7200 (sec)
    Verify return code: 0 (ok)
    Extended master secret: no
---

```

There is little information about this on the Internet, so I decided to write here.

**Comment by immortalua** [comment]
Solved the problem by changing /etc/nginx/conf.d/default.conf 

```nginx
ssl_protocols TLSv1 TLSv1.1 TLSv1.2 TLSv1.3;
ssl_ciphers 'DEFAULT:@SECLEVEL=0';
ssl_prefer_server_ciphers on;

```
You can also change the default template section in /app/nginx.tmpl and mount it in container by volume.

But in this case it doesn’t work TLSv1.3. I don't understand why

**Comment by buchdag** [comment, marked as answer]
That's being worked on in #2510

Edit: #2510 has been merged and this is solved.
